src/script/common/c_content.cpp

```diff
@@ -1770,6 +1770,32 @@ void read_json_value(lua_State *L, Json::Value &root, int index, u8 recursion)
 		size_t len;
 		const char *str = lua_tolstring(L, index, &len);
 		root = std::string(str, len);
+	} else if (is_real_lua_vector(L)) {
```

Comment:
This is inappropriate; vectors deserve no special handling in JSON. The provided implementation will decrease performance.